haddock-api/src/Haddock/Interface.hs

```diff
@@ -123,11 +124,12 @@ processModules verbosity modules flags extIfaces = do
   out verbosity verbose "Renaming interfaces..."
   let warnings = Flag_NoWarnings `notElem` flags
   dflags <- getDynFlags
-  interfaces'' <- withTimingM "renameAllInterfaces" (const ()) $ do
-    let (interfaces'', msgs) =
-           runErrMsgM $ mapM (renameInterface dflags (ignoredSymbols flags) links warnings) interfaces'
```

Comment:
Through some "watching verbose log output" sleuthing, I determined that generating Haddock for the work app was failing for OOM reasons on my 64GB laptop at this point.

The problem is that we're doing `runErrMsgM`, which accumulates everything into the `Writer ErrorMessages` before exiting. On the work app, this is around 4,000 modules, and some of these modules report a *huge* number of messages.

The `CPS` `Writer` *did* improve things considerably, but the overall behavior here is still bad.